shorebird_code_push/lib/src/shorebird_updater.dart

```diff
@@ -148,17 +148,5 @@ abstract class ShorebirdUpdater {
   /// * [isAvailable], which indicates whether the updater is available.
   /// * [checkForUpdate], which should be called to check if an update is
   ///   available before calling this method.
-  Future<void> update({UpdateTrack? track});
-}
-
-/// The track to check for updates on.
-enum UpdateTrack {
```

Comment:
We can remove the breaking-ness of this change by using an extension type and making UpdateTrack a string with these extra fields.

```
extension type UpdateTrack(String value) {
  static staging = "staging";
  static beta = "beta";
  static stable = "stable";

}
```

That should make this change non-breaking.

Reply:
I'll need to think about naming for a bit, but making this change non-breaking will probably best be solved by introducing a new pair of methods as counterparts to `checkForUpdate` and `update` that accept a `String` track name, and update the existing implementations to simply forward `track.name` to those new mehods.